src/client/TimeStamp.cpp

```diff
@@ -0,0 +1,39 @@
+#include "TimeStamp.hpp"
+#include "libftpp/utility.hpp"
+#include <algorithm>
+#include <climits>
+#include <ctime>
+
+int convertSecondsToMs(long timeoutSecs)
+{
+  timeoutSecs = std::max(timeoutSecs, 0L);
+  timeoutSecs =
+    std::min(timeoutSecs, static_cast<long>(INT_MAX) / MS_MULTIPLIER);
+  const long timeoutMs = timeoutSecs * MS_MULTIPLIER;
+  return static_cast<int>(timeoutMs);
+}
```

Comment:
There's a `numeric` header in libftpp.
```suggestion
int convertSecondsToMs(long timeoutSecs)
{
  const long timeoutMs = ft::mul_sat(timeoutSecs, MS_MULTIPLIER);
  return ft::numeric_cast<int>(timeoutMs, std::nothrow).value_or(INT_MAX);
}
```
`ft::mul_sat` does saturating arithmetic (exists also in the stdlib since C++26), so it clamps at the arguments' numeric boundaries.
`ft::numeric_cast` does checked casting to the type provided as a template argument. When passing `std::nothrow` it returns an `ft::expected` instead of throwing an exception, containing the result if the cast succeeded safely, or an error if not.
`ft::expected` provides the `value_or()` member function.